packages/clay-icon/src/ClayIcon.tsx

```diff
@@ -0,0 +1,42 @@
+/**
+ * © 2019 Liferay, Inc. <https://liferay.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import * as React from 'react';
+import classNames from 'classnames';
+
+export const ClayIconSpriteContext = React.createContext('');
+
+interface Props extends React.SVGAttributes<SVGSVGElement> {
+	className?: string;
+	spriteMap?: string;
+	symbol: string;
+}
+
+const ClayIcon: React.SFC<Props> = ({
+	className,
+	spriteMap,
```

Comment:
I think we can instantiate `useContext` here to remove the condition, I do not know if it would work or would break some Hooks rule.
```suggestion
	spriteMap = useContext(ClayIconSpriteContext),
```

Reply:
We could add a warning to help dev too, for example if spritemap is not passed via prop or does not have the provider in the application, it did not render the component and emit a error in the console for the developer.

That would be just a suggestion if we could adopt on Clay, use the https://www.npmjs.com/package/tiny-invariant lib, helps create those messages and it is also interesting to make these conditions, but we should ensure to add only in dev environment.